scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala

```diff
@@ -762,16 +776,37 @@ final class ScannerTokens(val tokens: Tokens)(implicit dialect: Dialect) {
                 val ok = next.is[KwCase] && sepRegions.headOption.contains(RegionCaseMark)
                 if (ok) emitIndent(sepRegions) else None
               case _: Colon =>
+                /**
+                 * Colon with NL can appear in several contexts:
+                 *   - after package/class/etc: handled with RegionColonEol
+                 *   - after variable and before its type:
+                 *     - within `def`, `val`, `var` declaration or definition: excluded with
+                 *       RegionDefMark
+                 *     - cast within an argument expression: doesn't allow newline as it can then be
+                 *       confused with a fewer-braces invocation
+                 *     - within self-type: not allowed by the compiler
+                 *   - after fewer-braces method call: will apply if not handled otherwise
+                 */
+                def couldBeFewerBraces(): Boolean =
+                  dialect.allowFewerBraces && !sepRegions.contains(RegionDefMark) &&
```

Comment:
`!sepRegions.headOption.contains(RegionDefMark) `

Shouldn't it be at the head?

Reply:
not really. here's an example:

```
def foo[A : Xxx](bar: Int): Int = 0
```

the first colon and second colon will have a RegionParen/RegionBracket first, before RegionDefMark, but they cannot start fewer braces.